Plugins/GitUIPluginInterfaces/Extensibility/ISettingsSource.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace GitUIPluginInterfaces.Extensibility
+{
+    public interface ISettingsSource
+    {
+        /// <summary>
+        /// Extracts the value with the specified key and converts it to type T.
+        /// </summary>
+        /// <typeparam name="T">The type to convert the value to.</typeparam>
+        /// <param name="key">The key of the setting value.</param>
+        /// <param name="defaultValue">The default value to use if no value is found.</param>
+        /// <returns>The converted value to T.</returns>
+        [return: NotNullIfNotNull("defaultValue")]
+        T? GetValue<T>(string key, T? defaultValue = default);
```

Comment:
```suggestion
        T? Get<T>(string key, T? defaultValue = default);
```

Reply:
I think "GetValue" will be clearer since we have a source.

`Source.GetValue` 

It's not as clear what we are getting as it seems to me.

`Source.Get`